src/test/scala/mesosphere/mesos/ConstraintsTest.scala

```diff
@@ -645,6 +645,30 @@ class ConstraintsTest extends UnitTest {
     }
   }
 
+  "IS operator" should {
+    "require that a value match exactly" in {
+      makeOffer("righthost.com") should meetConstraint(hostnameField, IS, "righthost.com")
+      makeOffer("wronghost.com") should meetConstraint(hostnameField, IS, "wronghost.com")
+    }
+  }
+
+  "IN operator" should {
+    "require that the offer value be in the comma delimited list of values" in {
+      makeOffer("host1") should meetConstraint(hostnameField, IN, "host1,host2")
+      makeOffer("host3") shouldNot meetConstraint(hostnameField, IN, "host1,host2")
+    }
+
+    "trims whitespace after the commas (but not before)" in {
+      makeOffer("host1") should meetConstraint(hostnameField, IN, "host1, host2")
+      makeOffer("host1") should meetConstraint(hostnameField, IN, "host1 , host2")
```

Comment:
Isn't here a whitespace before the comma? `host1` is not equal to `host1[ ]`. What am I missing?

Reply:
The separator is a comma optionally surrounded by whitespaces.

Reply:
err lol the test description is a lie 😢

Reply:
I don't believe Mesos constraints what values can be specified on an agent attribute value. A user could potentially put trailing or leading spaces in the value. That would be an awful thing to do, but, none-the-less, possible.

I think it is more likely that people accidentally put a space in comma-delimited list.

Reply:
Yes, in agreement here.

Reply:
@vinodkone corrected me; spaces _are explicitly_ not allowed in attribute values, and [they are removed](https://github.com/apache/mesos/blob/master/src/common/values.cpp#L561) by Mesos.